### PR TITLE
SONARPY-345 Parser should parse Python3 by default and fallback to Python2

### PR DESCRIPTION
**Pull request by pynicolas**
The other commit comes from #251

### CODE REVIEW

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonScanner.java [hunk 1]
```diff
@@ -104,17 +104,22 @@ private void scanFile(InputFile inputFile) {
...
-    pyFile.accept(pythonHighlighter);
-    pythonHighlighter.getNewHighlighting().save();
+    try {
+      PyFile pyFile = new org.sonar.python.frontend.PythonParser().parse(fileContent);
+      SubscriptionVisitor.analyze(checks.all(), visitorContext, pyFile);
+      PythonHighlighter pythonHighlighter = new PythonHighlighter(context, inputFile);
+      pyFile.accept(pythonHighlighter);
+      pythonHighlighter.getNewHighlighting().save();
+    } catch (RecognitionException e) {
+      // Ignore, we should not raise a parsing error twice (once for each parser)
```
Comment (andrea-guarino-sonarsource) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonScanner.java]: What do you think about adding `DEBUG` log message?

File: python-frontend/src/main/java/org/sonar/python/frontend/PythonParser.java [hunk 2]
```diff
@@ -63,7 +70,28 @@
...
+    file = parseAs(content, LanguageLevel.PYTHON27);
+    PsiErrorElement errorElement = errorElements(file).get(0);
+    if (errorElement == null) {
+      return file;
+    }
+    int lineNumber = new PythonTokenLocation(errorElement).startLine();
+    throw new RecognitionException(lineNumber, errorElement.getErrorDescription());
+  }
+
+  private JBIterable<PsiErrorElement> errorElements(PsiElement root) {
```
Comment (andrea-guarino-sonarsource) [inline python-frontend/src/main/java/org/sonar/python/frontend/PythonParser.java]: There are two code smells regarding `errorElements` and `parseAs` not being static

File: python-checks/src/main/java/org/sonar/python/checks/PrintStatementUsageCheck.java [hunk 3]
```diff
@@ -32,14 +36,35 @@
...
     context.registerSyntaxNodeConsumer(PyElementTypes.PRINT_STATEMENT, ctx -> {
       List<PyElement> expressions = PsiTreeUtil.getChildrenOfAnyType(ctx.syntaxNode(), PyExpression.class, PyPrintTarget.class);
       if (expressions.size() == 1 && expressions.get(0) instanceof PyParenthesizedExpression) {
         return;
       }
-      ctx.addIssue(ctx.syntaxNode().getFirstChild(), "Replace print statement by built-in function.");
+      ctx.addIssue(ctx.syntaxNode().getFirstChild(), MESSAGE);
+    });
+
+    context.registerSyntaxNodeConsumer(PyElementTypes.BINARY_EXPRESSION, ctx -> {
```
Comment (andrea-guarino-sonarsource) [inline python-checks/src/main/java/org/sonar/python/checks/PrintStatementUsageCheck.java]: Could you please add a comment explaining why we need this?

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonScanner.java (hunk 1 above)
Comment (pynicolas) [reply sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonScanner.java]: I had to rebase against the master. Now, we need both parsers to compute metrics. I had to [restructure the code](https://github.com/SonarSource/sonar-python/pull/252/commits/55903b135010295800d454f470d0766340f7e32a#diff-3939b2c0f41ee15dc69c575494495142).

File: python-frontend/src/main/java/org/sonar/python/frontend/PythonParser.java (hunk 2 above)
Comment (pynicolas) [reply python-frontend/src/main/java/org/sonar/python/frontend/PythonParser.java]: Fixed.

File: python-checks/src/main/java/org/sonar/python/checks/PrintStatementUsageCheck.java (hunk 3 above)
Comment (pynicolas) [reply python-checks/src/main/java/org/sonar/python/checks/PrintStatementUsageCheck.java]: Done.

**Comment by pynicolas** [comment]
I fixed the typo in the commit message.
I don't think we should create a separate ticket for `PrintStatementUsageCheck`: 
* there should be no change in behavior compared to what we had before
* I had to update `PrintStatementUsageCheck` precisely because of the first commit: in fact, unit tests are red after the 1st commit because of a [test file using a `print` statement](https://github.com/SonarSource/sonar-python/blob/master/sonar-python-plugin/src/test/resources/org/sonar/plugins/python/squid-sensor/file1.py#L10).

**Pull request merged**